test/Tests/src/System/File_Spec.enso

```diff
@@ -719,7 +723,16 @@ spec =
             filtered1 = root.list name_filter="**.txt" recursive=True . map .to_text
             filtered1.sort.should_equal (resolve ["sample.txt", "subdirectory/a.txt", "subdirectory/nested/b.txt"])
 
+            filtered1b = root.list name_filter="*.txt" recursive=True . map .to_text
+            filtered1b.sort.should_equal (resolve ["sample.txt", "subdirectory/a.txt", "subdirectory/nested/b.txt"])
+
             filtered2 = root.list name_filter="*/*/*" recursive=True . map .to_text
             filtered2.should_equal (resolve ["subdirectory/nested/b.txt"])
```

Comment:
This case looks slightly worrying. If I understand correctly what happens, it will be prepended with `**`. So now it will find all paths that have `>= 3` layers of nesting, whereas before this change it would only find paths that have `== 3` layers of nesting.

Looking at the call, I'd indeed expect `== 3` to be the correct behaviour here.

I appreciate the change, as indeed the other tests that you added show it was needed - specially the one just below - indeed if `recursive=True`, it's nice to find the file in all subdirectories. But it can break some usages for more advanced use cases. I'm not sure if we can do anything good about it though... It is not great that we cannot now use this method to find `== n` layers of nesting - we will always find arbitrarily nested files. On the other hand, maybe it's not that important - I think overall this change is a pragmatic improvement.